src/EthersSafe.ts

```diff
@@ -1,22 +1,23 @@
 import { Provider } from '@ethersproject/providers'
-import { BigNumber, ContractTransaction, Wallet } from 'ethers'
-import { GnosisSafe } from '../typechain'
+import { BigNumber, ContractTransaction, Signer } from 'ethers'
 import SafeAbi from './abis/SafeAbiV1-2-0.json'
 import ModuleManager from './managers/moduleManager'
 import OwnerManager from './managers/ownerManager'
 import Safe from './Safe'
-import { areAddressesEqual } from './utils'
+import { sameString } from './utils'
 import { generatePreValidatedSignature } from './utils/signatures'
 import { EthSignSignature, SafeSignature } from './utils/signatures/SafeSignature'
-import { SafeTransaction } from './utils/transactions'
+import { estimateGasForTransactionExecution } from './utils/transactions/gas'
+import SafeTransaction, { SafeTransactionDataPartial } from './utils/transactions/SafeTransaction'
+import { standardizeSafeTransaction } from './utils/transactions/utils'
 
 class EthersSafe implements Safe {
-  #contract: GnosisSafe
+  #contract: any
   #ethers: any
-  #ownerManager: OwnerManager
-  #moduleManager: ModuleManager
-  #provider: Provider
-  #signer?: Wallet
+  #ownerManager!: OwnerManager
```

Comment:
what is the exclamation mark for and why did it become required?

Reply:
I'm using the property `strictPropertyInitialization` in the tsconfig.json file, which enables strict checking of property initialization in classes.
Because there are some async tasks to check in the constructor of the class, I'm using a static and async method `create` to return an instance of the class. The property `strictPropertyInitialization` returns the error `Property '<name>' has no initializer and is not definitely assigned in the constructor.` because the constructor is not used to initialize them and the assertion operator (!) can be used to ingore that (ans still have the property `strictPropertyInitialization` everywhere else).
Here you have more info: https://www.typescriptlang.org/docs/handbook/2/classes.html#--strictpropertyinitialization

Reply:
Got it. I'd prefer usage of @ts-expect-error, though